src/canister_id.rs

```diff
@@ -4,54 +4,54 @@ use ic_agent::export::Principal;
 use crate::config::dns_canister_config::DnsCanisterConfig;
 
 /// A resolver for `Principal`s from a `Uri`.
-trait UriCanisterIdResolver: Sync + Send {
+trait UriResolver: Sync + Send {
     fn resolve(&self, uri: &Uri) -> Option<Principal>;
 }
 
-impl<T: UriCanisterIdResolver> UriCanisterIdResolver for &T {
+impl<T: UriResolver> UriResolver for &T {
     fn resolve(&self, uri: &Uri) -> Option<Principal> {
         T::resolve(self, uri)
     }
 }
 
 struct UriParameterResolver;
 
-impl UriCanisterIdResolver for UriParameterResolver {
+impl UriResolver for UriParameterResolver {
     fn resolve(&self, uri: &Uri) -> Option<Principal> {
         url::form_urlencoded::parse(uri.query()?.as_bytes())
             .find(|(name, _)| name == "canisterId")
             .and_then(|(_, canister_id)| Principal::from_text(canister_id.as_ref()).ok())
     }
 }
 
-impl UriCanisterIdResolver for DnsCanisterConfig {
+impl UriResolver for DnsCanisterConfig {
     fn resolve(&self, uri: &Uri) -> Option<Principal> {
         self.resolve_canister_id(uri.host()?)
     }
 }
 
 /// A resolver for `Principal`s from a `Request`.
-pub trait CanisterIdResolver<B>: Sync + Send {
+pub trait Resolver<B>: Sync + Send {
```

Comment:
The fact that you called the trait `Resolver` makes me think you're gonna be a great Go developer 😄